bugbug/similarity.py

```diff
@@ -175,3 +178,46 @@ def get_similar_bugs(self, query):
         return [
             self.bug_ids[ind] for ind in indices[0] if self.bug_ids[ind] != query["id"]
         ]
+
+
+class Word2VecWmdSimilarity(BaseSimilarity):
+    def __init__(self):
+        self.corpus = []
+        self.bug_ids = []
+        for bug in bugzilla.get_bugs():
+            self.corpus.append(text_preprocess(get_text(bug)))
+            self.bug_ids.append(bug["id"])
+        self.w2vmodel = Word2Vec(self.corpus, size=100, min_count=1)
+
+    def get_similar_bugs(self, query):
+        self.w2vmodel.init_sims()
+        words = text_preprocess(get_text(query))
+        words = [word for word in words if word in self.w2vmodel.wv.vocab]
+
+        all_distances = np.array(
+            1.0
+            - np.dot(
+                self.w2vmodel.wv.vectors_norm,
+                self.w2vmodel.wv.vectors_norm[
+                    [self.w2vmodel.wv.vocab[word].index for word in words]
+                ].transpose(),
+            ),
+            dtype=np.double,
+        )
+        distances = []
+        for i in range(len(self.corpus)):
+            idx = [self.w2vmodel.wv.vocab[word].index for word in self.corpus[i]]
+            if len(idx) != 0:
+                word_dists = all_distances[idx]
+                rwmd = max(
+                    np.sum(np.min(word_dists, axis=0)),
+                    np.sum(np.min(word_dists, axis=1)),
+                )
+
+                distances.append((self.bug_ids[i], rwmd))
+
+        distances.sort(key=lambda v: v[1])
+
```

Comment:
You should now apply real WMD, like at https://github.com/marco-c/crashsimilarity/blob/master/crashsimilarity/models/base.py#L165.